recipes/onnxruntime/all/conanfile.py

```diff
@@ -249,7 +250,10 @@ def package_info(self):
                 onnxruntime_libs.append("providers_xnnpack")
             self.cpp_info.libs = [f"onnxruntime_{lib}" for lib in onnxruntime_libs]
 
-        self.cpp_info.includedirs.append("include/onnxruntime/core/session")
+        if Version(self.version) < "1.16.0" or not self.options.shared:
+            self.cpp_info.includedirs.append("include/onnxruntime/core/session")
+        else:
+            self.cpp_info.includedirs.append("include/onnxruntime")
```

Comment:
Why `include/onnxruntime/core/session` if building a static library?

From https://github.com/microsoft/onnxruntime/blob/v1.16.0/cmake/onnxruntime.cmake#L123 seems to not be dependent on SHARED/STATIC ?

Reply:
the target onnxruntime only exist for shared library: https://github.com/microsoft/onnxruntime/blob/v1.16.0/cmake/CMakeLists.txt#L1616-L1621

So for static library, the install is not the same as for shared library.
I wonder if we should just rewrite includes in conan for static library in order to get the same layout & restricted set of includes ?

Reply:
Oh I see.

In my opinion the recipe should closely match what the upstream project intents to do otherwise it might be complicated to manage. 

Maybe something that could be done is to reflect ORT components with self.cpp_info.components (to easier identify where some include/dependency/etc comes from)

Reply:
If you do not build it as a shared library, you will be able to access all internal C/C++ functions. like onnxruntime::InferenceSession in onnxruntime/core/session/inference_session.cc.  We do not maintain any kinds of backward compatibility for them, but they have less overhead than the public C APIs in onnxruntime_c_api.h and they are more powerful.

Reply:
Thanks for the extra insight @snnn :)